docs/StorageManagement.md

```diff
@@ -0,0 +1,235 @@
+
+- Business logic should not be hard-coded in Fybrik. 
+
+- A single connection taxonomy should be used by modules, catalog conector and storage manager.
+
+- Deployment of FybrikStorageAccount CRD should be configurable - to be discussed: [issue 1717](https://github.com/fybrik/fybrik/issues/1717) 
+
+- Fybrik manages storage life cycle of temporary data copies.
+
+- A clear error indication should be provided if the requested storage type is not supported, or an operation has failed.
+
+- IT admin should be able to express config policies based on storage dynamic attributes (e.g., cost) as well as storage properties such as type, geography and others. 
+
+- Optimizer needs to ensure that the allocated storage type matches the connection that the module uses.
+
+- The selected storage should not necessarily match the source dataset connection in case of copying an existing asset.
+
+- [Future enhancement] The data user should be able to request a specific storage type, or to specify some of the connection properties (e.g., bucket name) inside FybrikApplication. 
+
+- The data user should be able to leave a choice of a storage type to Fybrik and organization policies.  
+
+## Goals 
+
+- Provide modules with a connection object for writing data
+
+- Share organization storage to store temporary or persistent data while hiding the details (credentials, server URLs, service endpoints, etc.) from the data user. This means that data users will store data in organization accounts created by IT administrators.  
+
+- Govern the use of the shared storage for the given workload according to the compliance, capacity and other factors.
+
+- Optimize the shared storage for the given workload (by cost, latency to the workload)
+
+- Manage storage life cycle of the shared storage (e.g., delete temporary data after not being used, delete empty buckets, etc.) 
+
+# High Level Design
+
+## Taxonomy and structures
+
+It is crucial that the modules and the storage allocation component have a base common connection structure.  Otherwise, the components will not work together correctly when deployed.  Thus, we propose the following:
+
+- **Connection** is defined in **base** taxonomy within Fybrik repository, as it is done today:
+
+```
+// +kubebuilder:pruning:PreserveUnknownFields
+// Name of the connection to the data source
+type Connection struct {
+	// Name of the connection to the data source
+	Name                 ConnectionType   `json:"name"`
+	AdditionalProperties serde.Properties `json:"-"`
+}
+```
+
+- Fybrik defines **taxonomy layers** with schema definition for supported connections (in pkg/taxonomy/layers). Quickstart deploys Fybrik using these layers. Users can add/replace layers when deploying Fybrik.
+
+In **Phase1** we define layers for all connection types that are supported today by open-source modules, such as `s3`, `db2`, `kafka`, `arrow-flight`. (Revisit taxonomy layers used by Airbyte module.)
```

Comment:
Do we really need a separate layer for each connection type?  Or can we keep them all in a single layer - which will be the open source community layer?

Reply:
It is a single layer. Breaking into smaller files is rather a technicality. I'll update.